event_sale_portal/controllers/portal.py

```diff
@@ -0,0 +1,60 @@
+from odoo import http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import content_disposition, request
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+
+
+class CustomerPortal(CustomerPortal):
+    @http.route(
+        [
+            "/my/orders/<int:order_id>/event/<int:event_id>/badge",
+            "/my/orders/<int:order_id>/event/<int:event_id>/badge/<int:reg_id>",
+        ],
+        type="http",
+        auth="public",
+        website=True,
+    )
+    def portal_order_event_badge(
+        self, order_id, event_id, reg_id=None, access_token=None, download=False, **kw
+    ):
+        """Download event registration badges from sale order portal page."""
+        try:
+            order_sudo = self._document_check_access(
+                "sale.order", order_id, access_token=access_token
+            )
+        except (AccessError, MissingError):
+            return request.redirect("/my")
+        # Get sale order downloadable registrations
+        registrations = order_sudo.event_registration_ids.filtered(
+            lambda r: (
+                r.state == "open"
+                and r.event_id.id == event_id
+                and r.event_id.portal_badge_download
+            )
+        )
+        if not registrations:
+            return request.redirect("/my")
+        # Check if registration_id belongs to the sale order
+        if reg_id is not None:
+            if reg_id not in registrations.ids:
```

Comment:
```suggestion
        if reg_id and reg_id not in registrations.ids:
```

Reply:
It's not the same though. See [line 42](https://github.com/OCA/event/pull/217/files#diff-e7b9aa0c19b7535023d1d615dd8cfc93c51f6f36b2f2d2480f0f798682e43d27R42)